index.js

```diff
@@ -32,7 +34,9 @@ module.exports = postcss.plugin('postcss-pseudo-classes', function (options) {
 
       rule.selectors.forEach(function (selector) {
         // Ignore some popular things that are never useful
-        if (blacklist[selector]) {
+        // ignore css pseudo class functions, like :host(...), :host-context(...)
+        // do not process selectors that start with blacklisted string
+        if (blacklist[selector] || selector.match(/.+?(?=\()/)) {
```

Comment:
Instead of making the check here we should do it here I think:

https://github.com/giuseppeg/postcss-pseudo-classes/blob/ce694b759b9f3fe8c9ce9bab193ba4356fc005f4/index.js#L61-L67

```diff
+ pseudo = pseudo.replace(/\(.*/g, '')

if (
  blacklist[pseudo] ||
  restrictTo &&
-  !restrictTo[pseudo.replace(/\(.*/g, '')]
+  !restrictTo[pseudo]
) {
  return pseudo;
}
```

Reply:
Hello @giuseppeg ,

I've added a test but I had to change the blacklist object to an Array of RegExp. 
Then I can compare the pseudo-string against banned regExp.

I'm not sure about to move the check below. I guess when it's filtered out above, you don't need the check again because it cannot reach that line in that case.

I also had to update ava dependencies and tests to run in my current node environment.

Reply:
Hi Philipp, I believe the first check was to skip simple selectors and return early. For example

```css
:host {
  color: red
}
```

Further down, where I added my suggestion, instead we process all of them. You can try to revert the changes you made to `index.js` and apply my suggestion: https://github.com/giuseppeg/postcss-pseudo-classes/pull/6#discussion_r344502959

If I am right the tests you've written should pass.

Reply:
Hi @giuseppeg ,

>  I believe the first check was to skip simple selectors and return early.

That's what I want, too. 

`:host`, `:host(...)` and `:host-context(...)` should be banned and skipped for generation.

Adding the check `pseudo.match(/.+?(?=\()/))` here
```js
if (
  blacklist[pseudo] ||
  pseudo.match(/.+?(?=\()/)) ||
  restrictTo &&
) {
  return pseudo;
}
```

will also ban all selectors with parentheses, like :nth-child(...) or :not(...) and already existing tests will fail.

Reply:
This is what I meant https://github.com/giuseppeg/postcss-pseudo-classes/pull/8
I merged that one but thank you so much for the contribution!!